scripts/ci/validate_ci_workflow.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import yaml
+import sys
+import glob
+from collections import Counter
+
+NOTEBOOK_MARKER = "# MARKER: list of all notebooks"
+NOTEBOOK_JOB = "notebooks"
+
+ALL_JOBS_PASSED_MARKER = "# MARKER: list of all jobs"
+ALL_JOBS_PASSED_JOB = "all-jobs-passed"
+
+WORKFLOW = ".github/workflows/ci.yml"
```

Comment:
This could potentially be generated via relative paths, if it's required to be run outside the folder

Reply:
Good idea, I'll use `__file__` and `os.path` to make this be relative to the script, rather than the PWD of the user who executes it.